frappe/public/js/frappe/form/sidebar/form_sidebar.js

```diff
@@ -144,18 +146,54 @@ frappe.ui.form.Sidebar = class {
 					fieldname: ["frequency"],
 				},
 				callback: function (res) {
-					me.sidebar
-						.find(".auto-repeat-status")
-						.removeClass("hidden")
-						.html(__("Repeats {0}", [__(res.message.frequency)]));
-					me.sidebar.find(".auto-repeat-status").on("click", function () {
+					let el = me.sidebar.find(".auto-repeat-status");
+					el.find("span").html(__("Repeats {0}", [__(res.message.frequency)]));
+					el.closest(".sidebar-section").removeClass("hidden");
+					el.show();
+					el.on("click", function () {
 						frappe.set_route("Form", "Auto Repeat", me.frm.doc.auto_repeat);
 					});
 				},
 			});
 		}
 	}
 
+	show_error_log_status() {
+		const me = this;
+		const filters = {
+			reference_doctype: this.frm.doc.doctype,
+			reference_name: this.frm.doc.name,
+		};
+		frappe.db.exists("Error Log", filters).then((exists) => {
```

Comment:
Could we set this information in doc onload to avoid subsequent HTTP calls?

Reply:
Of course we could! What's the pro/con?

My initial consideration was that this is vastly uncritical information, and it may eventually even be a costly query on a potentially large log table.

So I thought it would be better not to place that query into the critical path.

The downside of which is that now the delay is very noticeable (~500ms) until this information pops up in the sidebar.

I made this reasoning in isolation and on principle and did not compare it much to current behaviour or precedent of costly queries in the critical path. So my weights might have led to odd preferences and conclusions.

How would you reason, here?

Reply:
I don't think I saw the diff closely before adding that comment - I assumed it was regarding adding this for a specific DocType, hence the suggestion: onload. The `getdoc` endpoint seems more apt now that I look at it.

Reasoning: In a `getdoc` request, we bundle Comments, Communication, Versions & a bunch of other Documents. This seems like the appropriate place to bundle information regarding the existence of logs. Separating it out *feels* like an anti-pattern.

Although I do like your reasoning of not adding this to the critical path (Especially for setting the colour of an indicator), this is a simple existence check on Error & Webhook logs, it should be fairly lightweight on the typical Frappe site with the `tabComment` table being many times bigger than `tabError Log` and an empty `tabWebhook Log`. _(This is probably a dated impression, someone who is actively maintaining thousands of Frappe sites could comment)_ 

What is the cost of adding 2 additional API calls per Desk form load?  Is this cost acceptable when most of the forms loaded via Desk won't have a corresponding Error or Webhook log?

---

On another note, we can avoid the HTTP request in many cases by checking first if the user has read access to said Logs by `frappe.model.can_read("Error Log")`

Reply:
> Is this cost acceptable when most of the forms loaded via Desk won't have a corresponding Error or Webhook log?

Fortunately, the cardinality of the query to those log tables is good: 1. doctype 2. docname - I'll check if there are the appropriate indices set and add them otherwise. Then we should be ok, keeping in mind that all log tables are also regularily pruned after 30 days (and that is even user-configurable, if getting out of hand).

So your suggestions makes a ton of sense, actually. Thank you!